### ISSUE TITLE
Add goja as additional JSVM driver

### ISSUE THREAD

**Issue by joshblakeley** [TykTechnologies/tyk#1893]
**Do you want to request a *feature* or report a *bug*?**

Feature

**What is the current behavior?**
Currently we use Otto driver for our JSVM
We have option of using https://github.com/dop251/goja which claims a 6/7 times performance increase and a higher level of conformance to ECMAScript standards than Otto has.

It is clearly related to Otto and has a similar (but not identical) API.

I'd like us to give users the option of which JSVM is used when running virtual endpoints or javascript middleware, and this is a good candidate for our JSVM V2.

It can be chosen via a `jsvm_engine` option in config, where default value is "otto". Also JSVM plugins can also be loaded via bundles, and in manifest you specify "driver". It should accept `goja` value too.

**Comment by letzya** [comment]
+1 from another client

**Comment by l3ender** [comment]
Please keep open. We are using JSVM and would always welcome performance enhancements for it.

**Comment by andyo-tyk** [comment]
In our investigations into goja we identified no noticeable performance increase for the types of tasks that Tyk uses; for this reason we stopped any development towards switching the JSVM driver.

We are always on the look out for opportunities to improve the performance of Tyk, but this particular avenue is currently closed.

I will close this ticket to reduce noise, but the interest in performance enhancements in this area is noted for future product evolution.
